docs/configuration/config-file-reference.md

```diff
@@ -4067,6 +4067,15 @@ The `redis_config` configures the Redis backend cache.
 The `ruler_config` configures the Cortex ruler.
 
 ```yaml
+# Address of the Query Frontend service, in host:port format. Ruler queries to
+# Query Frontends via gRPC. If not set, ruler queries to Ingesters directly.
+# CLI flag: -ruler.frontend-address
+[frontend_address: <string> | default = ""]
+
+# HTTP timeout duration when querying to the Query Frontend.
+# CLI flag: -ruler.frontend-timeout
+[frontend_timeout: <duration> | default = 10s]
```

Comment:
This is something I am not sure about. If a rule is known to be slow, does a global frontend timeout make sense?
This changes the existing behavior as the current Rule evaluation doesn't have any timeout.

Reply:
I got it. Can I use -querier.timeout?